### PR TITLE
clj-kondo: Remove references to GraalVM derivation

### LINKED ISSUE #269029: clj-kondo attempting to access graalvm-ce
### Describe the bug

Install clj-kondo and the following error occurs:

```
$ nix develop --impure
error: output '/nix/store/y6h6cq22izlr5dag6yz5wcaf0vai1l30-clj-kondo-2023.10.20' is not allowed to refer to the following paths:
         /nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1
error: 1 dependencies of derivation '/nix/store/k8jfrm3ls00i8x58pwvwls80ibnh4r14-devenv-profile.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bzyb6xkp02qh2kgvp3y2l4n7cbwg851m-devenv-shell-env.drv' failed to build
```

No trouble installing clj-kondo a few weeks ago so could be a problem specific to the [latest release](https://github.com/NixOS/nixpkgs/pull/262474) if not my setup.

### Steps To Reproduce

<details><summary><code>flake.nix</code></summary>

``` nix
{
  description = "Example";

  inputs = {
    devenv.url =
      "github:cachix/devenv/9ba9e3b908a12ddc6c43f88c52f2bf3c1d1e82c1";

    flake-utils.url = "github:numtide/flake-utils";
    nixpkgs.url = "github:NixOS/nixpkgs";
  };

  outputs = { self, flake-utils, nixpkgs, devenv }@inputs:
    flake-utils.lib.eachDefaultSystem (system:
      let
        overlays = [
          (self: super: rec {
            nodejs = super.nodejs-18_x;
            pnpm = super.nodePackages.pnpm;
          })
        ];
        pkgs = import nixpkgs { inherit overlays system; };
      in {
        devShells.default = devenv.lib.mkShell {
          inherit inputs pkgs;
          modules = [
            ({ pkgs, ... }: {
              packages = with pkgs; [
                babashka
                clj-kondo
                clojure
                clojure-lsp
                nodejs
                pnpm
              ];

              # ----------------------------------------------------------------
              # Postgres

              services.postgres.enable = true;

              services.postgres.extensions = extensions: [
                extensions.pgvector
                extensions.postgis
              ];

              services.postgres.initialDatabases =
                [ { name = "dev"; } { name = "test"; } ];

              # So we can connect over TCP rather than via a socket.
              services.postgres.listen_addresses = "127.0.0.1";

              # ----------------------------------------------------------------
              # Processes

              env.DATABASE_URL =
                "jdbc:postgresql://127.0.0.1:5432/dev?user=dev&password=please";

              # Apple have taken port 5000 for Control Center.
              env.OVERMIND_PORT = "3000";

              # Use Overmind to workaround Tailwind's esbuild internals.
              process.implementation = "overmind";

              processes = {
              };
            })
          ];
        };
      });
}
```

</details> 

``` sh
nix develop --impure # because of devenv
```

### Expected behavior

clj-kondo should be installed.

### Notify maintainers

@r-ryantm @thiagokokada @bennyandresen @jlesquembre 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[satoshi@miner01:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - channels(jcf): `"darwin, home-manager"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

### See also

- [Slack thread in clojurians#clj-kondo](https://clojurians.slack.com/archives/CHY97NXE2/p1700663489243989)
- https://github.com/NixOS/nixpkgs/issues/95117

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by jcf**
## Description of changes

Remove references to GraalVM derivation. Fixes https://github.com/NixOS/nixpkgs/issues/269029.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Maintainers

@jlesquembre @bennyandresen @thiagokokada 

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/clj-kondo/default.nix [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
   };
 
-  extraNativeImageBuildArgs = [
-    "-H:+ReportExceptionStackTraces"
-    "--no-fallback"
-  ];
+  graalvmDrv = graalvmCEPackages.graalvm-ce;
+
+  extraNativeImageBuildArgs =
+    [ "-H:+ReportExceptionStackTraces" "--no-fallback" ];
```
Comment (jcf) [inline pkgs/development/tools/clj-kondo/default.nix:17]: Apologies for the unrelated changes to formatting. I have an over-enthusastic editor and limited mobility.

**Comment by jcf** [comment]
Successfully built on macOS!

<details><summary><code>nix-build -A clj-kondo</code></summary>

```
this derivation will be built:
  /nix/store/nhgs5a4g7zpm3597bxcfy7s7jjj12z7p-clj-kondo-2023.10.20.drv
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/cb1km89cdxr27njla01xc41lsdhb21mq-remove-references-to
copying path '/nix/store/cb1km89cdxr27njla01xc41lsdhb21mq-remove-references-to' from 'https://cache.nixos.org'...
building '/nix/store/nhgs5a4g7zpm3597bxcfy7s7jjj12z7p-clj-kondo-2023.10.20.drv'...
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
Apply jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/native-image.properties
Warning: Ignoring server-mode native-image argument --no-server.
Apply jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/reflect-config.json
Warning: The option '-H:Log=registerResource:' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:-CheckToolchain' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:Name=clj-kondo' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:IncludeResources=clj_kondo/impl/cache/built_in/.*' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it. The build output lists all active experimental options, including where they come from and possible alternatives. If you think an experimental option should be considered as stable, please file an issue.
Apply jar:file:///nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties
Apply jar:file:///nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties
Executing [
HOME=/homeless-shelter \
PATH=/nix/store/dwaqcdnxnzm4jlh6an8y3gfa9z89cfp7-darwin-cc/bin:/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin:/nix/store/cb1km89cdxr27njla01xc41lsdhb21mq-remove-references-to/bin:/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/bin:/nix/store/a5v30qll5i02vr9y97bk1rdx3mm6kvlm-clang-16.0.6/bin:/nix/store/h0fgpyxfav7ybnw5sdg8jpyv9f95x4w0-coreutils-9.3/bin:/nix/store/ydcx9rj4gf3dxm1kc0lx7l5d5vrjdwp2-cctools-binutils-darwin-wrapper-16.0.6-973.0.1/bin:/nix/store/vpnp0c420cjmx82g05jq3nm5skdrblvk-cctools-binutils-darwin-16.0.6-973.0.1/bin:/nix/store/h0fgpyxfav7ybnw5sdg8jpyv9f95x4w0-coreutils-9.3/bin:/nix/store/zr6klxfjzpdr2674ly1f4fix7ig57mjr-findutils-4.9.0/bin:/nix/store/94j70ziq1zp2mj0fs2fzi49j7q3mdhba-diffutils-3.10/bin:/nix/store/nddzbf5pifm77kksh25xwiy2g9kqqm9f-gnused-4.9/bin:/nix/store/98v0v3bwzgkcwsz4anjip62d2k3gz352-gnugrep-3.11/bin:/nix/store/blggp459hvq6swvr8nlxblkdn7ayw8y7-gawk-5.2.2/bin:/nix/store/g9z0hg6snhg5834p14yw4bx6ilcffr8b-gnutar-1.35/bin:/nix/store/6g3nq5hply5p2v3xzlg0i9qqgzr30w1d-gzip-1.13/bin:/nix/store/dq6dsprg57r5wqlc4yzcnkbfxdhfcd3f-bzip2-1.0.8-bin/bin:/nix/store/6kxdgmbgwdq4y42r9qhgdky312v5d7x0-gnumake-4.4.1/bin:/nix/store/x1xxsh1gp6y389hyl40a0i74dkxiprl7-bash-5.2-p15/bin:/nix/store/j34fimsig5s1zsf2f5dkw9a8sv5zpskg-patch-2.7.6/bin:/nix/store/96bznf6bjdghshgj3a2hm16gk3nngbcx-xz-5.4.4-bin/bin:/nix/store/ckbx2qla8zxic5k2d549qdhkbwrpa4q6-file-5.45/bin \
PWD=/private/tmp/nix-build-clj-kondo-2023.10.20.drv-0 \
USE_NATIVE_IMAGE_JAVA_PLATFORM_MODULE_SYSTEM=true \
/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/bin/java \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dtruffle.TrustAllTruffleRuntimeProviders=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dgraalvm.locatorDisabled=true \
-Dsubstratevm.IgnoreGraalVersionCheck=true \
--add-exports=java.base/com.sun.crypto.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.access=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.event=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.loader=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.logger=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.misc=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto \
--add-exports=java.base/jdk.internal.module=org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.org.objectweb.asm=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.perf=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.platform=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.ref=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=java.base/jdk.internal.reflect=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.vm.annotation=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.vm=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.invoke.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.net.www=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.net=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.nio.ch=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=java.base/sun.reflect.annotation=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.factory=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.reflectiveObjects=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.repository=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.scope=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.tree=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.jca=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.ssl=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.x509=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.text.spi=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.calendar=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.cldr=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.resources=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util=org.graalvm.nativeimage.builder \
--add-exports=java.management/com.sun.jmx.mbeanserver=org.graalvm.nativeimage.builder \
--add-exports=java.management/sun.management=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=jdk.internal.vm.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=jdk.internal.vm.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=jdk.internal.vm.compiler,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=jdk.internal.vm.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.events=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.jfc=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal=org.graalvm.nativeimage.builder \
--add-exports=jdk.management/com.sun.management.internal=org.graalvm.nativeimage.builder \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-XX:MaxRAMPercentage=46.530771255493164 \
-XX:GCTimeRatio=9 \
-XX:+ExitOnOutOfMemoryError \
-Djava.awt.headless=true \
'-Dorg.graalvm.vendor=GraalVM Community' \
-Dorg.graalvm.vendorurl=https://www.graalvm.org/ \
'-Dorg.graalvm.vendorversion=GraalVM CE 21.0.1+12.1' \
-Dorg.graalvm.version=23.1.1 \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Xshare:off \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-Dborkdude.dynaload.aot=true \
-Dsun.stdout.encoding=UTF-8 \
-Dsun.stderr.encoding=UTF-8 \
-Xmx6g \
--add-modules=ALL-DEFAULT \
--module-path \
/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/builder/svm.jar:/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/builder/native-image-base.jar:/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/builder/objectfile.jar:/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/builder/pointsto.jar \
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-imagecp \
/nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar \
-imagemp \
/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/library-support.jar \
-H:CLibraryPath=/nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/clibraries/darwin-aarch64 \
-H:Path@driver=/private/tmp/nix-build-clj-kondo-2023.10.20.drv-0 \
'-H:Name@manifest from file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar=fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone' \
'-H:Class@manifest from file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar=clj_kondo.main' \
'-H:Name@jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/native-image.properties=clj-kondo' \
'-H:ClassInitialization@jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/native-image.properties+api=com.fasterxml.jackson:build_time' \
'-H:ClassInitialization@jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/native-image.properties+api=com.github.javaparser:build_time' \
'-H:IncludeResources@jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/native-image.properties=clj_kondo/impl/cache/built_in/.*' \
'-H:Log@jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/native-image.properties=registerResource:' \
'-H:ReflectionConfigurationResources@jar:file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar!/META-INF/native-image/clj-kondo/clj-kondo/reflect-config.json+api=META-INF/native-image/clj-kondo/clj-kondo/reflect-config.json' \
-H:-CheckToolchain@user \
-H:Name@user=clj-kondo \
-H:MicroArchitecture@user+api=compatibility \
-H:+ReportExceptionStackTraces@user \
-H:FallbackThreshold@user+api=0 \
-H:ImageBuildID@driver=1b255fa6-869e-796b-9944-d19f4437e49d \
'-H:Features@jar:file:///nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///nix/store/yzdpwaz10q8lrc6hhdqjyzw7970j5pdl-graalvm-ce-21.0.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'clj-kondo' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
Warning: Feature class InitAtBuildTimeFeature is annotated with the deprecated annotation @AutomaticFeature. Support for this annotation will be removed in a future version of GraalVM. Applications should register a feature using the option --features=InitAtBuildTimeFeature
Registering packages for build time initialization: clojure, cognitect, bencode, pod.borkdude, sci, aaaa_this_has_to_be_first, cheshire, clj_kondo, babashka, datalog, edamame, clj_easy
[1/8] Initializing...                                                                                    (6.4s @ 0.23GB)
 Java version: 21.0.1+12, vendor version: GraalVM CE 21.0.1+12.1
 Graal compiler: optimization level: 2, target machine: compatibility
 C compiler: cc (null, null, 0.0.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - InitAtBuildTimeFeature
 - com.oracle.svm.thirdparty.gson.GsonFeature
------------------------------------------------------------------------------------------------------------------------
 3 experimental option(s) unlocked:
 - '-H:Name' (alternative API option(s): -o clj-kondo; origin(s): command line)
 - '-H:IncludeResources' (origin(s): 'META-INF/native-image/clj-kondo/clj-kondo/native-image.properties' in 'file:///nix/store/fdw6swz2qcrxqcxlbd4csry9rpi0jcas-clj-kondo-2023.10.20-standalone.jar')
 - '-H:-CheckToolchain' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 5.33GB of memory (8.3% of 64.00GB system memory, set via '-Xmx6g')
 - 10 thread(s) (100.0% of 10 available processor(s), determined at start)
[2/8] Performing analysis...  [***]                                                                     (21.7s @ 1.85GB)
   12,753 reachable types   (90.6% of   14,077 total)
   19,314 reachable fields  (42.8% of   45,107 total)
   50,959 reachable methods (59.6% of   85,521 total)
    2,149 types,   129 fields, and 1,552 methods registered for reflection
       62 types,    62 fields, and    55 methods registered for JNI access
        5 native libraries: -framework CoreServices, -framework Foundation, dl, pthread, z
[3/8] Building universe...                                                                               (3.3s @ 1.62GB)
[4/8] Parsing methods...      [**]                                                                       (2.0s @ 1.89GB)
[5/8] Inlining methods...     [***]                                                                      (1.0s @ 1.51GB)
[6/8] Compiling methods...    [****]                                                                    (15.5s @ 1.57GB)
[7/8] Layouting methods...    [**]                                                                       (2.7s @ 2.21GB)
[8/8] Creating image...       [***]                                                                      (8.1s @ 2.35GB)
  26.82MB (39.43%) for code area:    35,332 compilation units
  40.25MB (59.17%) for image heap:  504,972 objects and 146 resources
 973.52kB ( 1.40%) for other data
  68.02MB in total
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
  17.65MB f.10.20-standalone.jar                               7.97MB byte[] for code metadata
   7.08MB java.base                                            6.37MB java.lang.Class
   1.05MB svm.jar (Native Image)                               5.20MB java.lang.Object[]
 195.14kB jdk.crypto.ec                                        3.30MB byte[] for java.lang.String
 148.38kB java.desktop                                         2.92MB clojure.lang.PersistentHashMap$BitmapIndexedNode
 111.10kB java.logging                                         2.24MB java.lang.String
  75.15kB jdk.charsets                                         1.40MB clojure.lang.Symbol
  56.84kB org.graalvm.nativeimage.base                         1.07MB com.oracle.svm.core.hub.DynamicHubCompanion
  55.23kB java.xml                                           903.66kB clojure.lang.PersistentHashMap$INode[]
  51.78kB jdk.proxy3                                         785.08kB byte[] for embedded resources
 151.89kB for 11 more packages                                 8.13MB for 5974 more object types
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 INIT: Adopt '--strict-image-heap' to prepare for the next GraalVM release.
 HEAP: Set max heap for improved and more predictable memory usage.
------------------------------------------------------------------------------------------------------------------------
                        4.6s (7.5% of total time) in 90 GCs | Peak RSS: 3.70GB | CPU load: 5.29
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /private/tmp/nix-build-clj-kondo-2023.10.20.drv-0/clj-kondo (executable)
========================================================================================================================
Finished generating 'clj-kondo' in 1m 1s.
buildPhase completed in 1 minutes 4 seconds
installing
post-installation fixup
checking for references to /private/tmp/nix-build-clj-kondo-2023.10.20.drv-0/ in /nix/store/v34lrl32pkwva8x399zdf6rpl49pdi1g-clj-kondo-2023.10.20...
patching script interpreter paths in /nix/store/v34lrl32pkwva8x399zdf6rpl49pdi1g-clj-kondo-2023.10.20
stripping (with command strip and flags -S) in  /nix/store/v34lrl32pkwva8x399zdf6rpl49pdi1g-clj-kondo-2023.10.20/bin
/nix/store/v34lrl32pkwva8x399zdf6rpl49pdi1g-clj-kondo-2023.10.20
```

</details>

File: pkgs/development/tools/clj-kondo/default.nix (hunk 1 above)
Comment (thiagokokada) [reply pkgs/development/tools/clj-kondo/default.nix:19]: I really don't like `nixfmt` formatting but I think the reasoning is fine here.

**Review by thiagokokada — APPROVED** [review]

**Comment by jcf** [comment]
Thanks, @thiagokokada! 🙇
